wolfcrypt/src/cmac.c

```diff
@@ -369,7 +405,7 @@ int wc_AesCmacVerify(const byte* check, word32 checkSz,
     }
 
     XMEMSET(a, 0, aSz);
-    ret = wc_AesCmacGenerate(a, &aSz, in, inSz, key, keySz);
+    ret = wc_AesCmacGenerate_ex(a, &aSz, in, inSz, key, keySz, devId);
     compareRet = ConstantCompare(check, a, (int)min(checkSz, aSz));
```

Comment:
Fixed ret check. 

Regarding the cryptoCb, that is a great idea, but we would need to add additional info to the cryptoCb so that it could determine the difference between generate and verify. If the `type` field wasn't intended to be used to indicate the block cipher type, then we could use the field for that (as well as init/update/final). Otherwise we could add a new "operation" field to indicate this.